openwisp_utils/base.py

```diff
@@ -1,25 +1,50 @@
 import uuid
+from typing import Type
 
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
 from model_utils.fields import AutoCreatedField, AutoLastModifiedField
+from openwisp_utils.utils import get_random_key
+from openwisp_utils.validators import key_validator
 
 
-class TimeStampedEditableModel(models.Model):
-    """
-    An abstract base class model that provides self-updating
-    ``created`` and ``modified`` fields.
-    """
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    created = AutoCreatedField(_('created'), editable=True)
-    modified = AutoLastModifiedField(_('modified'), editable=True)
+def _key_model_factory(key_name: str) -> Type[models.Model]:
```

Comment:
I am not sure if this is a good solution to key problem. I made the factory, as in different repos the key is named differently. For example in [django-netjsonconfig it is `key`](https://github.com/openwisp/django-netjsonconfig/blob/master/django_netjsonconfig/base/device.py#L21-L26), but in [openwisp-radius it is called `token`](https://github.com/openwisp/openwisp-radius/blob/master/openwisp_radius/models.py#L142-L144)

Reply:
I think we're moving into too much abstraction territory with this one, you can remove it.

It's better to explicitly declare the key or token field each time for better readability and to make the code explicit therefore faster to inspect.